www/js/components/trades/trade-chart.directive.js

```diff
@@ -15,30 +15,263 @@ angular
 			restrict: 'E',
 			templateUrl: 'templates/components/trades/trade-chart.template.html',
 			link: function(scope, element) {
+
```

Comment:
Ideally we should refactor the code here amd move some of the code to chart service